shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala

```diff
@@ -732,7 +734,10 @@ object GlutenConfig {
         COLUMNAR_MEMORY_BACKTRACE_ALLOCATION.defaultValueString),
       (
         GLUTEN_COLUMNAR_TO_ROW_MEM_THRESHOLD.key,
-        GLUTEN_COLUMNAR_TO_ROW_MEM_THRESHOLD.defaultValue.get.toString)
+        GLUTEN_COLUMNAR_TO_ROW_MEM_THRESHOLD.defaultValue.get.toString),
```

Comment:
Let's check if spark.unsafe.sorter.spill.read.ahead.enabled is enabled or not. If not, close the spill read adhead buffer.

Reply:
FileInputStream implement decides it must read some buffer ahead even if spark.unsafe.sorter.spill.read.ahead.enabled is disabled. We could add a new class FileInputReader to read the file as need.

Reply:
what happens if we set the buffer size to 0?

Reply:
It does not check the config value reasonable.
It may cause dead cycle because it read file 0 byte one time.
 https://github.com/facebookincubator/velox/blob/main/velox/common/file/FileInputStream.cpp#L207

Reply:
Then let's ignore it for now and open an enhancement issue on it.

Reply:
Yes, added, https://github.com/facebookincubator/velox/issues/11673.
It won't affect us now, because Spark config requires the value is more than default value.
```
private[spark] val UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE =
    ConfigBuilder("spark.unsafe.sorter.spill.reader.buffer.size")
      .internal()
      .version("2.1.0")
      .bytesConf(ByteUnit.BYTE)
      .checkValue(v => 1024 * 1024 <= v && v <= MAX_BUFFER_SIZE_BYTES,
        s"The value must be in allowed range [1,048,576, ${MAX_BUFFER_SIZE_BYTES}].")
      .createWithDefault(1024 * 1024)
```

I will add the FileReadStream to Velox to support disable read ahead.